agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/telemetry/TelemetryChannel.java

```diff
@@ -364,19 +362,21 @@ private void handleStatsbeatOnResponse(
       String instrumentationKey, Consumer<Boolean> onFailure, OperationLogger operationLogger) {
 
     return error -> {
-      if (isStatsbeat
-          && (error instanceof UnknownHostException
-              || error instanceof UnresolvedAddressException)) {
-        // when sending a Statsbeat request and server returns an UnknownHostException, it's
-        // likely that it's using AMPLS. In that case, we use the kill-switch to turn off Statsbeat.
+      if (isStatsbeat && error instanceof Exception) {
+        // when sending a Statsbeat request and server returns an Exception, it's
+        // likely that it's using AMPLS or other private endpoints. In that case, we use the
+        // kill-switch to turn off Statsbeat.
+        // TODO (heya) track Statsbeat exception count via a new Statsbeat metric
```

Comment:
I don't follow the TODO, can you elaborate a bit?

Reply:
do you see a value to know when people are running into exception when sending statsbeat? here we just shut it down and forget about it. that is what I meant to track exception count in this case for statsbeat specific requests.

Reply:
how would we track it if we're shutting down statsbeat?

Reply:
Updated todo.